console/console-init/ui/src/modules/device/components/DeviceFilterCriteria/DeviceFilterCriteria.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { IDropdownOption, DropdownWithToggle } from "components";
+import {
+  InputGroup,
+  TextInput,
+  Button,
+  Grid,
+  GridItem
+} from "@patternfly/react-core";
+import { MinusCircleIcon } from "@patternfly/react-icons";
+
+export enum operator {
+  lte = "lessthanequalto",
+  gte = "greaterthanequalto",
+  lt = "lessthan",
+  gt = "greaterthan",
+  eq = "equalto",
+  neq = "notequalto"
+}
+
+export const operatordropdownOptions: IDropdownOption[] = [
+  { key: operator.eq, value: operator.eq, label: "=" },
+  { key: operator.neq, value: operator.neq, label: "≠" },
+  { key: operator.gte, value: operator.gte, label: "≥" },
+  { key: operator.gt, value: operator.gt, label: ">" },
+  { key: operator.lte, value: operator.lte, label: "≤" },
+  { key: operator.lt, value: operator.lt, label: "<" }
+];
+
+export interface IDeviceFilterCriteria {
+  parameter: string;
+  operator: string;
+  value: string;
+  key: string;
+}
+interface IDeviceFilterCriteriaProps {
+  criteria: IDeviceFilterCriteria;
+  deleteCriteria: (criteria: IDeviceFilterCriteria) => void;
+  setCriteria: (criteria: IDeviceFilterCriteria) => void;
+}
+const DeviceFilterCriteria: React.FunctionComponent<IDeviceFilterCriteriaProps> = ({
+  criteria,
+  deleteCriteria,
+  setCriteria
+}) => {
+  const onChangeParameter = (value: string) => {
+    const criteriaObj = criteria;
+    criteria.parameter = value;
+    setCriteria(criteriaObj);
+  };
+
+  const onChangeValue = (value: string) => {
+    const criteriaObj = criteria;
+    criteria.value = value.trim();
+    setCriteria(criteriaObj);
+  };
+
+  const onSelectOperator = (value: string) => {
+    const criteriaObj = criteria;
+    criteria.operator = value;
+    setCriteria(criteriaObj);
+  };
+
+  return (
```

Comment:
i would like to suggest use destructing to avoid use same object again and again.
const {key,parameter,value,operator}= criteria;

It will also help in case you object name 'criteria' change then you will need to make change at one place otherwise you will have to change each and every places.